feature time: Add stakeholder requirements (#1600)
eclipse-score/score: docs/requirements/stakeholder/index.rst

   * Diagnostic trouble codes
   * Diagnostic jobs

Time
----

Time synchronization to internal (in-vehicle) sources
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

.. stkh_req:: In-Vehicle Time base Synchronization
   :id: stkh_req__time__in_vehicle_synchronization
   :reqtype: Functional
   :security: NO
   :safety: QM
   :rationale: Enables the system to compare events chronologically.
   :status: valid

   The software platform shall provide a time synchronization framework to synchronize its clock

>    with external to ECU, but internal to vehicle time sources.

Author: ```suggestion
   to Time Master within the vehicle
```

.. stkh_req:: In-Vehicle Time base API
   :id: stkh_req__time__in_vehicle_time_base_api
   :reqtype: Functional
   :security: NO
   :safety: QM
   :rationale: Enables an application to correlate its data with a vehicle-internal time reference
   for event timestamping and chronological events comparison.
   :status: valid

>    The software platform shall provide an API to read the synchronized time from Local Time Slave.

Author: ```suggestion
   The software platform shall provide an API to access the synchronized vehicle time.
```

.. stkh_req:: In-Vehicle Time synchronization precision
   :id: stkh_req__time__in_vehicle_time_sync_precision
   :reqtype: Functional
   :security: NO
   :safety: QM
   :rationale: Enables the system to compare high frequent events chronologically.
   :status: valid

   The software platform shall synchronized the local time base with Global Time within a defined
   precision, based on the system setup.

   Note:

> 

Author: * the particular thresholds depends on the system setup

Author: +

   * for AVB nodes the maximum difference in the synchronized time between two AVB ports/ECUs while
   both are in state SYNC, and could be 1 usec

>    * for non-AVB node the precision is released, comparing to AVB ones

Author: add the upper limit: 200-500 usec

Author: +

.. stkh_req:: In-Vehicle Time synchronization local clock adaptations
   :id: stkh_req__time__in_vehicle_time_sync_local_clock_adaptations
   :reqtype: Functional
   :security: NO
   :safety: QM
   :rationale: Enables the system to compare high frequent events chronologically.
   :status: valid

>    The software platform shall keep the threshold for all local clock adaptations (including

Author: The software platform shall make  local clock adaptations (...) within the predefined time window

When it starts?
Does it make sense tomove it to the feature level?

   timestamping, converting and so on) which take part in time communication, based on the system setup.
   Note:

>    * for AVB nodes the threshold the threshold might be 100 nsec

Author: ```suggestion
   * for AVB nodes the threshold might be 100 nsec
```

System time provider
^^^^^^^^^^^^^^^^^^^^

.. stkh_req:: High precision Clock API
   :id: stkh_req__time__external_high_precision_clock_api
   :reqtype: Functional
   :security: YES
   :safety: QM
   :rationale: Enables an application to get the current system time, which is essential for time-sensitive operations and event scheduling,
   via common, mockable and standardized API.
   :status: valid

   The software platform shall provide the APIs to read the current high precision clock from the system time provider in nanoseconds.

   Note: to which clock the high precision clock is mapped, depends on the system design.

.. stkh_req:: Monotonic Clock API
   :id: stkh_req__time__external_monotonic_clock_api
   :reqtype: Functional
   :security: YES
   :safety: QM
   :rationale: Enables an application to get the current system time, which is essential for time-sensitive operations and event scheduling,
   via common, mockable and standardized API.
   :status: valid

   The software platform shall provide the APIs to read the current monotonic clock from the system time provider.

>    Note: to which clock the monotonic clock is mapped, depends on the system design.

Author: 1. synchronization to Time master with external to ECU (shall support)
3. synchronization to external time base ECU (UTC) 8shall support)
4. provide tha access to differnet timebases (or per timebase)

all the rest move to feature reqs.

   * Diagnostic trouble codes
   * Diagnostic jobs

Time Synchronization
--------------------

.. stkh_req:: Time Synchronization
   :id: stkh_req__time__synchronization
   :reqtype: Functional

>    :security: NO

Reviewer: I'm not sure what could happen if the time sychronization can be manipulated... Therefore I would expect that time sychronization is subject of necessary security measures...

Author: For synchronization to external sources - definitely.
For in-vehicle,I would say - no

   :safety: QM
   :rationale: Enables the system to synchronize its internal clock with external time sources, ensuring the data aging across distributed ECUs.
   :status: valid

>    The software platform shall provide time synchronization framework to synchronize its clock with external time sources.

Reviewer: ```suggestion
   The software platform shall provide a time synchronization framework to synchronize its clock with external time sources.
```

Author: Fixed

   The software platform shall provide a method and interface to enable
   debugging of the software on target and in vehicle.

> .. stkh_req:: Fake implementation for application testing

Reviewer: ```suggestion
.. stkh_req:: Mockup implementation for application testing
```

Let's use the more development like term "mockup" or  "Mocking". At least in the requirement headline.

>    :id: stkh_req__dev_experience__fake_public_apis

Reviewer: ```suggestion
   :id: stkh_req__dev_experience__mockup_public_apis
```
